src/goto-symex/goto_symex_state.cpp

```diff
@@ -280,14 +280,12 @@ static void set_l2_indices(
   ssa_expr.set_level_2(level2.current_count(ssa_expr.get_identifier()));
 }
 
-void goto_symex_statet::rename_level0(exprt &expr, const namespacet &ns)
+exprt goto_symex_statet::rename_level0(exprt expr, const namespacet &ns)
```

Comment:
The parameter should be `const exprt &expr`

Reply:
In the follow up PR I will make this return another type, and then having the argument passed by copy allow to move the argument, potentially saving a copy. I also refactored the code so that this will not get called recursively.